theano/gpuarray/opt.py

```diff
@@ -235,10 +250,10 @@ def local_opt(node):
                     elif isinstance(new_op, (tuple, list)):
                         return [safe_to_cpu(o) for o in new_op]
                     else:  # suppose it is a variable on the GPU
-                        return [host_from_gpu(new_op)]
+                        return [new_op.transfer('cpu')]
             return False
         local_opt.__name__ = maker.__name__
-        return local_optimizer(OP)(local_opt)
+        return register_topo('fast_compile')(local_optimizer(OP)(local_opt))
```

Comment:
I'm not sure I got you. Do you mean in the current setting, Equilibrium optimizer will perform an inefficient backward pass ? That was why @lamblin wanted the backward pass to be registered as an out to in Topological Optimizer, that will go from output to the input, to lift off these operations, 
`GpuFromHost(op(x)) => gpu_op(GpuFromHost(x))`